corehq/motech/repeaters/repeater_generators.py

```diff
@@ -261,6 +266,103 @@ def get_test_payload(self, domain):
         )
 
 
+@attr.s
+class CaseTypeReferralConfig(object):
+    #if false, listed_properties is a whitelist
+    use_blacklist = attr.ib()
+    listed_properties = attr.ib()
+    constant_properties = attr.ib()
+
+
+class ReferCasePayloadGenerator(BasePayloadGenerator):
+
+    def get_payload(self, repeat_record, payload_doc):
+
+        def _update_case_id(original_case_id, case_id_map):
```

Comment:
This doesn't exactly update a case id. Maybe `_update_case_id_map` or `_get_new_case_id`?

Reply:
good catch, it used to modify the case being passed in. I changed the functionality but didnt change the name accordingly